src/index.tsx

```diff
@@ -5,19 +5,36 @@ import App from "./components/App";
 import ErrorBoundary from "./components/ErrorBoundary";
 import { Provider } from "react-redux";
 import { store } from "./redux/store";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import NotFound from "./components/common/Error/NotFound";
 
 import "@fontsource/montserrat";
 
 import "./index.css";
 
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+    errorElement: <NotFound />,
+    children: [
+      {
+        index: true,
+        path: "search",
+        element: <App />,
+      },
+    ],
+  },
+]);
```

Comment:
First, can the App component be a child component of itself?

Secondly, is the children property needed at all? Do you have nested routes?

If you read the documentation again, you may notice such a property as [basename](https://reactrouter.com/en/main/routers/create-browser-router#basename)

```suggestion
const router = createBrowserRouter(
  [
    {
      path: "/",
      element: <App />,
      errorElement: <NotFound />,
    },
  ],
  {
    basename: "/search",
  }
);
```

Now you can remove useEffect in `App.tsx`

Reply:
my bad, some data was cached on my side, that is why solution worked for me, but the right one is:
```javascript
import { Navigate } from "react-router-dom";

const router = createBrowserRouter([
  {
    path: "/",
    element: <Navigate replace to="/search" />,
    errorElement: <NotFound />,
  },
  {
    path: "search",
    element: <App />,
  },
]);
```